python-client/tests/pytorch/test_newspaper_classification_pytorch_dataset.py

```diff
@@ -0,0 +1,123 @@
+import os
+import re
+
+import httpretty
+import pandas as pd
+import torch
+from scipy import special
+from torch import nn
+from torch.utils.data import Dataset as torch_dataset
+from torchtext.data.functional import to_map_style_dataset
+from torchtext.data.utils import get_tokenizer
+from torchtext.datasets import AG_NEWS
+from torchtext.vocab import build_vocab_from_iterator
+
+import tests.utils
+from giskard import PyTorchModel, Dataset
+from giskard.client.giskard_client import GiskardClient
+
+train_iter = AG_NEWS(split='train')
+test_iter = AG_NEWS(split='test')
+ag_news_label = {1: "World",
+                 2: "Sports",
+                 3: "Business",
+                 4: "Sci/Tec"}
+num_class = len(ag_news_label.keys())
+
+tokenizer = get_tokenizer('basic_english')
+
+
+def yield_tokens(data_iter):
+    for _, text in data_iter:
+        yield tokenizer(text)
+
+
+vocab = build_vocab_from_iterator(yield_tokens(train_iter), specials=["<unk>"])
+vocab.set_default_index(vocab["<unk>"])
+
+text_pipeline = lambda x: vocab(tokenizer(x))
+
+
+class PandasToTorch(torch_dataset):
```

Comment:
this class can be reused in `test_newspaper_classification_pytorch_custom_model.py`

Reply:
in the `test_newspaper_classification_pytorch_custom_model.py` example, it shouldn't be used, as I display the case of defining a custom model that takes df directly from it's `clf_predict` (that is the same previous implementation of `prediction_function`)